Makefile

```
@@ -105,11 +111,16 @@ $(BUILD)/test.out: $(BUILD)/test.js | $(BUILD)
 	node $< | tee $@.tmp # Store output in a temp file in case of a failure.
 	mv $@.tmp $@ # Move the output where it belongs.
 
+$(FORMAT)/%.purs.formatted: %.purs $(NODE_MODULES)
+	npx purty --write $<
+	@mkdir -p $$(dirname $@)
+	@touch $@
+
 $(NODE_MODULES): package.json
 	npm install
 	touch $@
 
-$(OUTPUT)/Main/index.js: $(EXAMPLES) $(SRCS) $(BOWER_COMPONENTS) $(NODE_MODULES) | $(BUILD)
+$(OUTPUT)/Main/index.js: format $(EXAMPLES) $(SRCS) $(BOWER_COMPONENTS) $(NODE_MODULES) | $(BUILD)
 	npx psa $(PSA_ARGS) $(RTS_ARGS) $(DEPS) $(EXAMPLES) $(SRCS)
 
 $(OUTPUT)/Test.Main/index.js: $(EXAMPLES) $(SRCS) $(TESTS) $(BOWER_COMPONENTS) $(NODE_MODULES) | $(BUILD)
```

```suggestion
$(OUTPUT)/Test.Main/index.js: $(FORMAT_EXAMPLES) $(FORMAT_SRCS) $(FORMAT_TESTS) $(BOWER_COMPONENTS) $(NODE_MODULES) | $(BUILD)
```

Similar to above, if we talk about which things we really need to depend on, it should keep things more accurate.

---

Side note, I'm not entirely sure why it's trying to remove a blank space at the start of the line. Hopefully this doesn't break the file.